File: 13.py
ls = []
with open("13.txt") as f:
    for line in f:
        l, d = line.strip().split(": ")
        ls.append((int(l), int(d)))

def severity(delay, ls):
    sev = 0
    for l in ls:
        if not (l[0]+delay) % (l[1]*2-2):
            sev += (l[0]+delay) * l[1]
    return sev

def catched(delay, ls):
    for l in ls:
        if not (l[0]+delay) % (l[1]*2-2):
            return True
    return False

print("Part 1:",severity(0,ls))
delay = 0
while catched(delay,ls):
    delay += 1
print("Part 2:", delay)

File: 13.jl
ls = open("13.txt") do f
    ls = []
    for line in eachline(f)
        l, d = (x->parse(Int, x)).(split(line, ": "))
        push!(ls, (l,d))
    end
    ls
end

function severity(delay, ls)
    sev = 0
    for (l, d) in ls
        if mod(l+delay, (d*2-2)) == 0
            sev += (l+delay) * d
        end
    end
    sev
end

function catched(delay, ls)
    for (l, d) in ls
        if mod(l+delay, (d*2-2)) == 0
            return true
        end
    end
    return false
end

function p2(ls)
    delay = 0
    while catched(delay, ls)
        delay +=1
    end
    delay
end

println("Part 1: ", severity(0, ls))
println("Part 2: ", p2(ls))
